test/Dialect/SPIRV/Serialization/function_call.mlir

```diff
@@ -0,0 +1,68 @@
+// RUN: mlir-translate -serialize-spirv %s | mlir-translate -deserialize-spirv | FileCheck %s
+
+spv.module "Logical" "GLSL450" {
+  spv.globalVariable @var1 : !spv.ptr<!spv.array<4xf32>, Input>
+  func @fmain() -> () {
+    %0 = spv.constant 16 : i32
+    %1 = spv._address_of @var1 : !spv.ptr<!spv.array<4xf32>, Input>
+    // CHECK: {{%.*}} = spv.FunctionCall @f_0({{%.*}}) : (i32) -> i32
+    %3 = spv.FunctionCall @f_0(%0) : (i32) -> i32
+    // CHECK: spv.FunctionCall @f_1({{%.*}}, {{%.*}}) : (i32, !spv.ptr<!spv.array<4 x f32>, Input>) -> ()
+    spv.FunctionCall @f_1(%3, %1) : (i32, !spv.ptr<!spv.array<4xf32>, Input>) ->  ()
+    // CHECK: spv.FunctionCall @f_2() : () -> ()
+    spv.FunctionCall @f_2() : () -> ()
+    // CHECK: spv.FunctionCall @f_3({{%.*}}) : (!spv.ptr<!spv.array<4 x f32>, Input>) -> !spv.ptr<!spv.array<4 x f32>, Input>
+    %4 = spv.FunctionCall @f_3(%1) : (!spv.ptr<!spv.array<4xf32>, Input>) -> !spv.ptr<!spv.array<4xf32>, Input>
+    // CHECK: spv.FunctionCall @f_3({{%.*}}) : (!spv.ptr<!spv.array<4 x f32>, Input>) -> !spv.ptr<!spv.array<4 x f32>, Input>
+    %5 = spv.FunctionCall @f_3(%4) : (!spv.ptr<!spv.array<4xf32>, Input>) -> !spv.ptr<!spv.array<4xf32>, Input>
+    spv.Return
+  }
+  func @f_0(%arg0 : i32) -> i32 {
+    spv.ReturnValue %arg0 : i32
+  }
+  func @f_1(%arg0 : i32, %arg1 : !spv.ptr<!spv.array<4xf32>, Input>) -> () {
+    spv.Return
+  }
+  func @f_2() -> () {
+    // CHECK: spv.FunctionCall @f_2() : () -> ()
+    spv.FunctionCall @f_2() : () -> ()
+    spv.Return
+  }
+  func @f_3(%arg0 : !spv.ptr<!spv.array<4xf32>, Input>) -> !spv.ptr<!spv.array<4xf32>, Input> {
+    spv.ReturnValue %arg0 : !spv.ptr<!spv.array<4xf32>, Input>
+  }
+
+  func @f_4() -> () {
+    %count = spv.constant 16 : i32
+    // CHECK: spv.FunctionCall @f_loop_with_function_call({{%.*}}) : (i32) -> ()
```

Comment:
Don't need this one? It's already covered in fmain.

Reply:
I was thinking to check that Function call can work under control flow ops such as loops.

Reply:
But that's for the next function?

Reply:
@antiagainst your are right! Thanks!